tests/test_doc_examples.py

```diff
@@ -0,0 +1,63 @@
+# Copyright 2019 Capital One Services, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import sys
+import yaml
+from parameterized import parameterized
+import itertools
+
+from c7n.resources import load_resources
+from c7n.exceptions import PolicyValidationError
+from c7n.provider import resources
+from .common import BaseTest
+
+
+def get_doc_examples():
+    policies = []
+    for key, v in resources().items():
+        #filters
+        for k, filter in v.filter_registry._factories.items():
+                if filter.__doc__:
+                    split_doc = [x.split('\n\n ') for x in
+                                 filter.__doc__.split('yaml')]  # split on yaml and new lines
+                    for item in itertools.chain.from_iterable(split_doc):
+                        if 'policies:\n' in item:
+                            policies.append((item, key, filter.__name__))
+        #actions
+        for k, action in v.action_registry._factories.items():
+            if action.__doc__:
+                split_doc = [x.split('\n\n ') for x in
+                             action.__doc__.split('yaml')]  # split on yaml and new lines
+                for item in itertools.chain.from_iterable(split_doc):
+                    if 'policies:\n' in item:
+                        policies.append((item, key, action.__name__))
+
+    return policies
+
+
+def custom_test_name(testcase_func, param_num, param):
+    return "test_docs_%s_%s" % (
+        param.args[1],
+        param.args[2]
+    )
+
+
+class DocExampleTest(BaseTest):
+
+    policies_in_docs = get_doc_examples()
+
+    @parameterized.expand(policies_in_docs, name_func=custom_test_name)
+    def test_doc_examples(self, policy, module, name):
+        parsed_policy = yaml.load(policy)
+        policy = self.load_policy(parsed_policy["policies"][0])
```

Comment:
this should iterate the policies, ie. if there is more than one.

Reply:
there should only be one policy at a time. the yaml.load creates a list of len 1 for each policy. i handle parsing multiple policies in a docstring already in `get_doc_examples`

Reply:
the parse only catches a particular format for separating multiple policies namely double line return and space, in checking this i found that that meant it didn't catch a few policy examples due to lack of iteration here.

also should be using yaml.safe_load or the utility load func we have in c7n.utils

Reply:
hmm do you have examples of which policies didnt get caught. and the `\n\n` catches the policies that have one empty line afterwards, since the first \n just refers to the last policy line.

Reply:
aws.account ServiceLimit                                                                               
aws.account XrayEncrypted
aws.account SetXrayEncryption
aws.elb SSLPolicyFilter
aws.s3 BucketEncryption